informer/error.go

```diff
@@ -2,8 +2,25 @@ package informer
 
 import (
 	"github.com/giantswarm/microerror"
+	"github.com/prometheus/client_golang/prometheus"
 )
 
+var alreadyRegisteredError = microerror.New("duplicate metrics collector registration attempted")
```

Comment:
The convention here is to have an error message like `already registered`. When you want to provide more information use the microerror annotation via `Maskf`. This is all described in https://github.com/giantswarm/fmt/blob/master/go/general_style_guide.md.

Reply:
I check this against the actual error returned from Prometheus though:
https://github.com/giantswarm/operatorkit/blob/72c70b0b4b10aee62fbc7e142097934034aaf3a7/informer/error.go#L17

So I assumed the message has to match the one provided by prom here:
https://github.com/prometheus/client_golang/blob/f1323f902ca878b3a87e1a76458be272fe1efe0d/prometheus/registry.go#L187

Or is this not necessary?

Reply:
This is not how the error matching works. Four things.

- Error matching for us works via type assertions. Error messages are totally irrelevant there. 
- We need to match against the cause of the error because we do the microerror magic. 
- We want to check if the error is of the type of this fancy prometheus error. If the type is what we expect the matcher should just return true. 
- We have conventions to make life easier. The conventions here are about the existence of the private error and public error matcher to have this error/matcher pairs so we can always just work with them. So the matcher has to check against our private error type as well and if this matches we return true too.

Reply:
I did this now hopefully correct. :man_facepalming: 
Thanks for the help.